third_party/move/tools/move-unit-test/src/test_runner.rs

```diff
@@ -75,23 +77,43 @@
     cost_schedule
 }
 
-/// Setup storage state with the set of modules that will be needed for all tests
+/// Setup storage state with the set of modules that will be needed for all tests.
 fn setup_test_storage<'a>(
     modules: impl Iterator<Item = &'a CompiledModule>,
-) -> Result<InMemoryStorage> {
-    let mut storage = InMemoryStorage::new();
+    genesis_state: Option<ChangeSet>,
+) -> Result<(InMemoryStorage, LocalModuleBytesStorage)> {
+    let mut resource_storage = InMemoryStorage::new();
+    let mut module_storage = LocalModuleBytesStorage::empty();
+
     let modules = Modules::new(modules);
-    for module in modules
-        .compute_dependency_graph()
-        .compute_topological_order()?
-    {
-        let module_id = module.self_id();
+    let dependency_graph = modules.compute_dependency_graph();
+    for module in dependency_graph.compute_topological_order()? {
         let mut module_bytes = Vec::new();
         module.serialize_for_version(Some(module.version), &mut module_bytes)?;
-        storage.publish_or_overwrite_module(module_id, module_bytes);
+        resource_storage.publish_or_overwrite_module(module.self_id(), module_bytes.clone());
+        module_storage.add_module_bytes(
+            module.self_addr(),
+            module.self_name(),
+            module_bytes.into(),
+        );
     }
 
-    Ok(storage)
+    if let Some(genesis_state) = genesis_state {
+        // In V2 loader implementation, we differentiate between modules and resources, so
```

Comment:
Is the problem that the move unit test setup stores modules and resources differently? Seems to me that they should be all viewed as bytes so I don't quite get why we need different logic here?